content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+                        name: cert
+                  volumes:
+                  - name: kubesvc-plugin-vol
+                    emptyDir: {}
+                  - name: cert
+                    secret:
+                      secretName: <your-secret-name>
+{{< /prism >}}
+
+
+### Configure the plugin
+
+Configure the plugin to use the mounted certs in the `agent-plugin/config.yaml` file. Note that `certificateChain`, `trustCertCollection`, and `privateKey` values must in `file:///filepath/filename` format.
+
+{{< prism lang="yaml" line="12-20" >}}
+apiVersion: spinnaker.armory.io/{{< param "operator-extended-crd-version" >}}
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        # See https://docs.armory.io/docs/installation/armory-agent/agent-options/
+        kubesvc:
+          cluster: redis
+          grpc:
+            server:
+              security:
+                enabled: true
+                certificateChain: file:<path-to-agent.crt>
+                trustCertCollection: file:<path-to-agent.crt>
+                privateKey: file:<path-to-agent.key>
+                clientAuth: REQUIRE
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-plugin-options.md" >}} page for additional options.
+
+## Agent service configuration
+
+### Create a secret
+
+Create a secret in the target cluster namespace where the Agent resides.
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration to mount the certs. @TODO what file is this in?
+
+If you have a custom CA, you need to mount the cert into the known trusted cert location:  `/etc/ssl/cert.pem`. Golang apps in Alpine use `/etc/ssl/cert.pem` as the source of a trusted CA.
+
+  1. Obtain a copy of the `cert.pem` file from a running Agent.
+  2. Append your custom CA onto the copied `cert.pem`  (for example `cat ca.crt >> cert.pem`)
+  3. Mount `cert.pem` to Agent deployment
+
+@TODO is this config correct? do the "this didn't work" sections work now or not?
+
+{{< prism lang="yaml"  >}}
+spec:
+  template:
+    spec:
+      containers:
+        volumeMounts:
+        - mountPath: /opt/kubesvc/cert
+          name: armoryagentcert
+        - mountPath: /opt/spinnaker/config
+          name: volume-kubesvc-config
+        - mountPath: /opt/kubesvc/cacert  # this didn't work as of Nov 2020
+          name: clouddrivercacert         # this didn't work as of Nov 2020
+      volumes:
+      - name: armoryagentcert
+        secret:
+          secretName: <agent-secret-name>
+      - name: clouddrivercacert         # this didn't work as of Nov 2020
+        secret:                         # this didn't work as of Nov 2020
+          secretName: <your-secret-name> # this didn't work as of Nov 2020
+      - name: certpem
+        secret:
+          secretName: ??? @TODO what secret is this?
```

Comment:
It's unclear what "didn't work as of Nov 2020." I think this was copied from some internal docs which may have had some specific requirements for files already existing? Not sure. In any case, several secrets are created earlier in these docs. Those secrets are:
A CA certificate  (See comment on line 153)
A clouddriver cert
An agent cert

Each of those is used in lines 168-176. As long as the secret names in this file match the names of the secrets that were created, this should work just fine. 

I think you should remove the "As of.. this didn't work" comments. If it configured properly, it works. To be fair, I think there was a bug or two that may have prevented some of this behavior from working properly, but the issues have been resolved.

Reply:
Depending on how they configured things, this may have been an issue (resolved): https://github.com/armory-io/kubesvc/pull/151/files